in-progress/8025-publishing-tx-objects/design.md

```diff
@@ -48,27 +69,159 @@ To the DA oracle contract, the proposer will publish the following data:
   - Tx2
   - ...
 
+> Most of the global variables could be directly populated by L1, but providing it makes it a lot clearer what the reason of failure is if such a one is encountered. Also the `inHash` can be populated by L1.
 
-To the rollup contract, the proposer will publish:
+#### DA Oracle Changes
 
-- CallData
-  - numTxs
-  - txsHash (a commitment to first nullifiers of all transactions in the (eventual) block)
-  - inHash
-  - GlobalVariables
-    - slotNumber
-    - timestamp
-    - coinbase
-    - feeRecipient
+When the data is to be submitted as part of the same transaction as the block proposal to the L1 contract, we have two variations possible:
 
-#### DA Oracle Changes
+1. Remove the data availability oracle in its whole, and simply use the versioned KZG hashes directly
+2. Save the versioned KZG hashes in the data availability oracle
 
-Keeps track of the versioned KZG hashes that were present in a transaction.
+Solution 1 is significantly cheaper as it don't update storage. Furthermore, by requiring that the versioned KZG hashes are part of the header provided, we can show the exact KZG hashes using just the transaction and the commitment to the block hash, see `BlockLog.blockHash`.
 
 #### Rollup Contract Changes
+> There is probably some 1-off from the `count` `block_number` like values.
+
+```python
+struct State:
+  pending_proposal_count: uint64;
```

Comment:
Storing the `count` versus `index`/`number`/`height` is really messing with me.

Reply:
Will look at trying it to indexes only to not mess with you 👍